src/components/sliding-panel.tsx

```diff
@@ -0,0 +1,121 @@
+import { FC, useRef, useEffect, ReactNode } from 'react';
+import { createPortal } from 'react-dom';
+import cn from 'classnames';
+
+import { Button, CloseIcon } from './index';
+
+import '../styles/components/sliding-panel.scss';
+
+type LRBelowHeader = {
+  /**
+   * Where the sliding panel should appear
+   */
+  position: 'left' | 'right';
+  /**
+   * Horizontal position of the arrow. If the panel appears below the page header.
+   * Also works as a flag to display the arrow and display below the header
+   */
+  arrowX?: number;
+};
+
+type TBSlidingPanel = {
+  /**
+   * Where the sliding panel should appear
+   */
+  position: 'top' | 'bottom';
+  arrowX?: never;
+};
+
+type SlidingPanelProps = {
+  /**
+   * What happens when close is triggered. Responsability of the user of the compoent
+   */
+  onClose: (arg: void) => void;
+  /**
+   * Size of the panel once opened
+   */
+  size?: 'small' | 'medium' | 'large' | 'full-screen';
+  /**
+   * Title of the panel
+   */
+  title?: ReactNode;
+  /**
+   * Add a close button in the panel header
+   */
+  withCloseButton?: boolean;
+  /**
+   * Should the panel be scrollable if the content is taller than the panel?
+   */
+  yScrollable?: boolean;
+  className?: string;
+} & (LRBelowHeader | TBSlidingPanel);
+
+const SlidingPanel: FC<SlidingPanelProps> = ({
+  children,
+  onClose,
+  position,
+  size = 'medium',
+  title,
+  withCloseButton = false,
+  className,
+  yScrollable = false,
+  arrowX,
+}) => {
+  const node = useRef<HTMLDivElement>(null);
+
+  const onCloseRef = useRef(onClose);
+  onCloseRef.current = onClose;
+
+  useEffect(() => {
+    const handleClickOutside = (e: MouseEvent) => {
+      if (node?.current && !node.current.contains(e.target as Node)) {
+        e.stopPropagation();
+        e.preventDefault();
+        onCloseRef.current();
+      }
+    };
+
+    document.addEventListener('click', handleClickOutside, true);
+    return () => {
+      document.removeEventListener('click', handleClickOutside, true);
+    };
+  }, []);
+
+  return createPortal(
+    <div
+      data-testid="sliding-panel"
+      className={cn(
+        'sliding-panel',
+        `sliding-panel--${position}`,
+        `sliding-panel--${position}--${size}`,
+        Number.isFinite(arrowX) && `sliding-panel--${position}--below-header`,
+        className
+      )}
+      style={{ overflowY: yScrollable ? 'auto' : 'initial' }}
+      ref={node}
+    >
+      {(title || withCloseButton) && (
+        <div className="sliding-panel__header">
+          {title && <h1 className="small">{title}</h1>}
+          {withCloseButton && (
+            <div className="sliding-panel__header__buttons">
+              <Button variant="tertiary" onClick={() => onCloseRef.current()}>
+                <CloseIcon />
+              </Button>
+            </div>
+          )}
+          {arrowX && (
```

Comment:
I'm not sure it would make sense? We were looking at this together with @aurel-l and decided to go for that. However using it in uniprot-website now, I think the arrow should be offset by half of its width (so the point is at `arrowX`) so will make that change.

Reply:
I was in favour of using `Number.isFinite` to make sure that someone re-reading the code later knows what we meant and doesn't assume we forgot the case where x=0. I know it's not likely to happen that x=0 but still

Reply:
Noticed that with `arrowX := 0` and without the `Number.isFinite` check a `0` appears on the righthand side of the title
<img width="819" alt="Screenshot 2021-06-22 at 14 15 33" src="https://user-images.githubusercontent.com/4357962/122930928-4f909d80-d364-11eb-92b8-6de3ed19b97a.png">

Reply:
Had missunderstood the purpose of `isFinite`, should now be working as expected and allowing 0